src/plugins/vis_type_vega/public/components/vega_info_message.tsx

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useMemo } from 'react';
+import { parse } from 'hjson';
+import { ExperimentalMapLayerInfo, shouldShowMapLayerInfo } from './experimental_map_vis_info';
+import {
+  DeprecatedHistogramIntervalInfo,
+  shouldShowDeprecatedHistogramIntervalInfo,
+} from './deprecated_interval_info';
+
+import type { Vis } from '../../../visualizations/public';
+import type { VegaSpec } from '../data_model/types';
+
+const parseSpec = (spec: string) => {
+  if (spec) {
+    try {
+      return parse(spec, { legacyRoot: false, keepWsc: true });
+    } catch (e) {
+      // spec is invalid
+    }
+  }
+};
+
+const InfoMessage = ({ spec }: { spec: string }) => {
+  const vegaSpec: VegaSpec = useMemo(() => parseSpec(spec), [spec]);
```

Comment:
If spec is invalid you try anyway to call `shouldShowMapLayerInfo` and `shouldShowDeprecatedHistogramIntervalInfo` which lead to crush app.

![image](https://user-images.githubusercontent.com/16915480/131361550-584ac218-b059-435a-b0be-e2ae760033ab.png)